Doc/library/sqlite3.rst

```diff
@@ -364,6 +364,143 @@ Module functions
 
+.. function:: Binary(data)
+
+   Create a Binary object to handle binary data in SQLite.
+
+   :param data:
+       The binary data to be encapsulated. This can be any object supporting the
+       Python buffer protocol, such as bytes, bytearray, or memoryview.
+   :type data: buffer-compatible object
+
+   The :func:`Binary` function encapsulates binary data to ensure proper handling
+   by SQLite. It is used to signal to the sqlite3 module that the data should be
+   treated as a BLOB (Binary Large Object) rather than text.
+
+   **Purpose and Functionality:**
+
+   - **Data Encapsulation**: Ensures the data is treated as binary and not as text.
+   - **Buffer Protocol Support**: Accepts objects that support the Python buffer protocol.
+   - **Type Signaling**: Explicitly signals that the data should be treated as binary.
+
+   **Internal Workings:**
+
+   - Creates an instance of the Binary class, a subclass of bytes.
+   - Flags the data for special handling as binary by the sqlite3 module.
+   - Ensures data is bound as a BLOB in SQL queries.
+
+   **Use Cases:**
+
+   - **Storing Raw Binary Data**: Ideal for files, images, or any raw binary data.
+   - **Preserving Exact Byte Sequences**: Prevents encoding/decoding errors.
+   - **Working with Non-Text Data**: Ensures data isn't subjected to text-based operations.
+
+   **Behavior in Queries:**
+
+   - **INSERT or UPDATE**: Stores Binary objects as BLOB data.
+   - **SELECT**: Retrieves BLOB data as Binary objects, convertible to bytes.
+
+   **Performance Considerations:**
+
+   - Using Binary() for large data can be memory-intensive.
+   - For very large objects, consider using SQLite's incremental or BLOB I/O interfaces.
+
+   **Compatibility and Portability:**
+
+   - Ensures consistent behavior across Python implementations and SQLite versions.
+   - Helps in writing portable code for binary data handling.
+
+   **Error Handling:**
+
+   - Raises :exc:`TypeError` if input doesn't support the buffer protocol.
+   - Handle potential :exc:`MemoryError` for very large binary objects.
+
+   **Best Practices:**
+
+   - Use Binary() to explicitly mark data as binary.
+   - Be mindful of memory usage for large binary objects.
+   - Consider using with prepared statements for efficiency.
+
+   By using :func:`Binary`, developers can ensure robust handling of binary data in SQLite,
+   preventing data corruption and enhancing application reliability.
+
+   **Examples:**
+
+   Basic example of storing and retrieving binary data:
+
+   .. code-block:: python
+
+       import sqlite3
+
+       # Create a connection and cursor
+       conn = sqlite3.connect(':memory:')
+       cursor = conn.cursor()
+
+       # Create a table
+       cursor.execute('''
+       CREATE TABLE files (
+           id INTEGER PRIMARY KEY,
+           name TEXT,
+           data BLOB
+       )
+       ''')
+
+       # Insert binary data
+       with open('example.png', 'rb') as file:
```

Comment:
In examples, I'd prefer *not* to create myself an additional file just to see how it could work. Just create some binary data yourself, like `bytes("abc", "utf-8")`.